samples/javascript_nodejs/21.corebot-app-insights/index.js

```diff
@@ -95,6 +105,10 @@ server.post('/api/messages', (req, res) => {
     });
 });
 
+// Enable the Application Insights middleware, which helps correlate all activity
+// based on the incoming request.
+server.use(restify.plugins.bodyParser());
```

Comment:
1. Are you using the requests body anywhere? I was unable to find where there value was being used and if it is not used then why add this body parser code?
2. Assuming this is needed, how is it related to telemetry?

Reply:
I believe this is required in order to ensure the incoming post is parsed correctly so we have access to it later on in the pipeline.  See here where req.body is used https://github.com/microsoft/botbuilder-js/blob/94edbde2d40fb834c28f02475341dabd544f6acb/libraries/botbuilder-applicationinsights/src/applicationInsightsTelemetryClient.ts

and also this doc for restify which talks about what the body parser does...
http://restify.com/docs/plugins-api/#bodyparser